src/main.rs

```diff
@@ -267,17 +268,77 @@ fn build(config: &Config) {
     };
 }
 
+fn static_file_handler(dest: &str, req: Request, res: Response) {
+    let path = PathBuf::from(format!("{}{}", dest, req.uri));
```

Comment:
After some digging, I did this:

```rust
fn static_file_handler(dest: &str, req: Request, res: Response) {
    let req_path = match req.uri {
        RequestUri::AbsolutePath(p) => p,
        _ => {
            error!("Request is not of the format AbsolutePath(p)");
            std::process::exit(1);
        },
    };

    let path = PathBuf::from(dest).join(&req_path[1..]);

    // ...
}
```

I tried many times, with paths that should fail and with correct paths, but all of them fell into `AbsolutePath(p)`. I'm still unsure of how to deal with the other cases, though.

Also, instead of relying in `format!`, this rids `p` of its first character (in all the tests I made, `p` started with `/`, and that was causing `join` to think `p` was an absolute path, so, instead of appending, it replaced the contents of `path` with `p`), and relies on `join` to get the local path to the file.